ciao-controller/api/api.go

```diff
@@ -1065,6 +1181,127 @@ func volumeAction(bc *Context, w http.ResponseWriter, r *http.Request) (Response
 	return Response{http.StatusBadRequest, nil}, err
 }
 
+func createInstance(c *Context, w http.ResponseWriter, r *http.Request) (Response, error) {
+	vars := mux.Vars(r)
+	tenant := vars["tenant"]
+
+	defer r.Body.Close()
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return Response{http.StatusBadRequest, nil}, err
+	}
+
+	var req CreateServerRequest
+
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		return Response{http.StatusBadRequest, nil}, err
+	}
+
+	resp, err := c.CreateServer(tenant, req)
+	if err != nil {
+		return errorResponse(err), err
+	}
+
+	return Response{http.StatusAccepted, resp}, nil
+}
+func listInstanceDetails(c *Context, w http.ResponseWriter, r *http.Request) (Response, error) {
+	vars := mux.Vars(r)
+	tenant := vars["tenant"]
+
+	values := r.URL.Query()
+
+	var workload string
+
+	// if this function is called via an admin context, we might
+	// have {workload} on the URL. If it's called from a user context,
+	// we might have workload as a query value.
+	workload, ok := vars["workload"]
+	if !ok {
+		if values["workload"] != nil {
```

Comment:
Not sure this check is sufficient to prevent a panic.  It should be 

if len(values["workload"]) > 0

[]string{} is a non nil slice with no elements.